File: backend/app/schemas/company.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, model_validator


class CompanyCreate(BaseModel):
    name: str
    eik: str
    address: str
    is_vat_registered: bool
    vat_exempt_reason: str | None = None
    invoice_number_prefix: str | None = None

    @model_validator(mode="after")
    def check_vat_exempt_reason(self) -> "CompanyCreate":
        if not self.is_vat_registered and not self.vat_exempt_reason:
            raise ValueError(
                "vat_exempt_reason е задължително, ако фирмата не е регистрирана по ДДС"
            )
        return self


class CompanyUpdate(CompanyCreate):
    pass


class CompanyRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    eik: str
    address: str
    is_vat_registered: bool
    vat_exempt_reason: str | None
    invoice_number_prefix: str | None
    logo_filename: str | None
    next_invoice_number: int
    created_at: datetime
    updated_at: datetime


class CompanyMemberRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    email: str
    full_name: str | None
    role: str | None


class AddMemberRequest(BaseModel):
    email: str

File: frontend/src/types/company.ts
export interface Company {
  id: number
  name: string
  eik: string
  address: string
  is_vat_registered: boolean
  vat_exempt_reason: string | null
  logo_filename: string | null
  next_invoice_number: number
  created_at: string
  updated_at: string
}

export interface CompanyInput {
  name: string
  eik: string
  address: string
  is_vat_registered: boolean
  vat_exempt_reason: string | null
}

export interface CompanyMember {
  id: number
  email: string
  full_name: string | null
  role: string | null
}
